pkg/controllers/tagging/tagging_controller.go

```diff
@@ -281,3 +327,31 @@ func (tc *Controller) enqueueNode(obj interface{}, action func(node *v1.Node) er
 	tc.workqueue.Add(item)
 	klog.Infof("Added %s to the workqueue", item)
 }
+
+func (tc *Controller) isTaggingRequired(node *v1.Node) bool {
+	labels := convertTagsToLabels(tc.tags)
+
+	if node.Labels != nil {
+		for key := range labels {
+			if _, ok := node.Labels[key]; !ok {
+				// If label key is missing in node, we tag and label the node.
+				return true
+			}
+		}
+	} else {
+		return true
+	}
+
+	return false
+}
+
+func convertTagsToLabels(tags map[string]string) map[string]string {
+	// Kubernetes label names can't contain ':' whereas it is very common in tags so convert that to '.' instead.
+	// Also, Kubernetes label keys and values shouldn't be more than 63 characters long so we ignore the values
+	// while labeling so that we can allow tag values to be longer than that.
+	labels := map[string]string{}
+	for key := range tags {
+		labels[strings.ReplaceAll(key, ":", ".")] = ""
```

Comment:
So this will mean there is a possible collision where `tagkey:xyz` and `tagkey.xyz` will use the same label, so we should probably at least print a warning in the logs.  Is that the only difference in allowed characters?

Reply:
It is hard to find out exactly what the difference is because not all AWS services have the same tag restrictions and this can change. I am now wondering if instead of us converting the tags to labels and using that for check, maybe we should have a tagging controller specific label which we can use to identify if a node was already processed?